curvefs/src/client/dentry_cache_manager.cpp

```diff
@@ -61,10 +60,14 @@ CURVEFS_ERROR DentryCacheManagerImpl::GetDentry(uint64_t parent,
             return CURVEFS_ERROR::OK;
         }
     }
-    curve::common::WriteLockGuard lg(mtx_);
-    bool ok = dCache_->Get(key, out);
-    if (ok) {
-        return CURVEFS_ERROR::OK;
+
+    NameLockGuard lock(nameLock_, key);
+    {
+        curve::common::ReadLockGuard lg(mtx_);
+        bool ok = dCache_->Get(key, out);
```

Comment:
LRU cache already lock put & get in its own logical，maybe mtx_ here is not needed？

Reply:
> LRU cache already lock put & get in its own logical，maybe mtx_ here is not needed？

Yes, the lock into LRU can protect operations on cache, the namelock can sequence the operations on the same one.